### PR TITLE
Prevent OverflowErrors in exponential_backoff()

### PR DESCRIPTION
**Pull request by kreuzert**
We start JupyterLabs on a slurm based batch system. Therefore, JupyterLabs may take a while to start. This lead to OverflowErrors in exponential_backoff(). 

This small patch will prevent crashes while waiting for JupyterLabs.

### CODE REVIEW

**Comment by manics** [comment]
Do you know roughly what value triggers OverflowError in practice? I'm wondering if it'd be simpler to bound `dt` by a fixed max-value instead.

File: jupyterhub/utils.py [hunk 1]
```diff
@@ -171,7 +171,15 @@ async def exponential_backoff(
         # add some random jitter to improve performance
         # this prevents overloading any single tornado loop iteration with
         # too many things
-        dt = min(max_wait, remaining, random.uniform(0, start_wait * scale))
+        # Except Overflow Error. Otherwise start_wait * scale will be too big at some point.
```
Comment (manics) [inline jupyterhub/utils.py:174]: Ignore my previous comment. Could this be simplified to something like:
```py
if dt < max_wait:
    dt = min(max_wait, remaining, random.uniform(0, start_wait * scale))
    scale *= scale_factor
```
else leave`dt` unchanged?

File: jupyterhub/utils.py [hunk 2]
```diff
@@ -171,7 +171,15 @@ async def exponential_backoff(
...
+        try:
+            if scale == 0:
+                dt = min(max_wait, remaining))
```
Comment (consideRatio) [inline jupyterhub/utils.py:177]:
```suggestion
                dt = min(max_wait, remaining)
```

syntax error

File: jupyterhub/utils.py (hunk 1 above)
Comment (kreuzert) [reply jupyterhub/utils.py:174]: Thanks for your response. I think any code snippet that prevents `scale *= scale_factor` from growing at some point will be fine.

**Comment by minrk** [comment]
Thanks!

**Pull request merged**